File: app/Domain/Messages/MessageId.php
<?php

namespace App\Domain\Messages;

class MessageId
{
    /** @var string */
    private $id;

    /**
     * @return MessageId
     */
    public static function generate()
    {
        return new MessageId(uniqid());
    }

    /**
     * @param string $id
     */
    public function __construct($id)
    {
        $this->id = $id;
    }

    /**
     * @return string
     */
    public function getId()
    {
        return $this->id;
    }
}

File: Mixter.Domain/Core/Messages/MessageId.cs
using System;

namespace Mixter.Domain.Core.Messages
{
    public struct MessageId
    {
        public MessageId(string id)
            : this()
        {
            Id = id;
        }

        public string Id { get; private set; }

        public static MessageId Generate()
        {
            return new MessageId(Guid.NewGuid().ToString());
        }

        public override string ToString()
        {
            return Id;
        }
    }
}
